src/Execution/ValuesResolver.php

```diff
@@ -75,49 +91,76 @@ public function coerceArgumentValues($objectType, $field, array $variableValues
                     $defaultValue
                 );
             } else {
-                $coercedValues[$argumentName] = $this->coerceValueForASTNode(
-                    $argumentNode,
-                    $argumentType,
-                    $argumentName,
-                    $variableValues
-                );
+                $coercedValue = null;
+
+                try {
+                    $coercedValue = $this->coerceValueFromAST($argumentNode->getValue(), $argumentType,
+                        $variableValues);
+                } catch (CoercingException $ex) {
```

Comment:
I just wonder should we move the logic in this `else` block to `coerceValueFromAST` method?

Reply:
The reason I changed this is because that method is used elsewhere and there we’re not coercing args.